### PR TITLE
Add retries on both get and posts to Percy's API

### PR DESCRIPTION
**Pull request by timhaines**

### CODE REVIEW

File: percy/connection.py [hunk 1]
```diff
@@ -7,6 +9,28 @@ def __init__(self, config, environment):
         self.config = config
         self.user_agent = str(UserAgent(config, environment))
 
+    def _requests_retry_session(
+        self,
+        retries=3,
+        backoff_factor=0.3,
+        method_whitelist=['HEAD', 'GET', 'POST', 'PUT', 'DELETE', 'OPTIONS', 'TRACE'],
+        status_forcelist=(500, 502, 503, 504, 520),
```
Comment (teeberg) [inline percy/connection.py]: for what it's worth, one of our requests to percy failed with a `524 Server Error: Origin Time-out`

Comment (timhaines) [reply percy/connection.py]: Ha, thanks @teeberg!  I knew there was another 52X code I needed to add, but hadn't looked up the precise one yet.  Will add 524 now.

Comment (teeberg) [reply percy/connection.py]: that was fast :-) thanks!

**Comment by timhaines** [comment]
@fotinakis Have exercised this manually, and seen that it does indeed retry on the 5XX responses I've targeted.  Will retry posts up to 3 times with exponential backoff.

**Pull request merged**